File: frontend/src/types/queue.ts
/**
 * Queue-related TypeScript types
 */

export type TaskStatus =
  | "pending"
  | "running"
  | "completed"
  | "failed"
  | "cancelled";

export interface QueueTask {
  id: number;
  session_id: string;
  status: TaskStatus;
  priority: number;
  start_agent: string;
  resume_from: string | null;
  created_at: string;
  started_at: string | null;
  completed_at: string | null;
  error_message: string | null;
}

export interface QueueStatus {
  is_processing: boolean;
  current_task: {
    id: number;
    session_id: string;
    started_at: string | null;
  } | null;
  pending_count: number;
}

export interface QueuePosition {
  session_id: string;
  position: number | null;
  total_pending: number;
}

File: backend/app/schemas/queue.py
"""
Pydantic schemas for queue-related API endpoints
"""

from datetime import datetime
from typing import Optional, Dict, Any

from pydantic import BaseModel


class QueueStatusResponse(BaseModel):
    """Response schema for queue status."""
    is_processing: bool
    current_task: Optional[Dict[str, Any]] = None
    pending_count: int


class QueueTaskResponse(BaseModel):
    """Response schema for queue task."""
    id: int
    session_id: str
    status: str
    priority: int
    start_agent: str
    resume_from: Optional[str]
    created_at: datetime
    started_at: Optional[datetime]
    completed_at: Optional[datetime]
    error_message: Optional[str]

    class Config:
        from_attributes = True


class QueuePositionResponse(BaseModel):
    """Response schema for queue position query."""
    session_id: str
    position: Optional[int]  # None if not in queue, 0 if running
    total_pending: int
